--- ./src/Makefile.in
+++ ./src/Makefile.in
@@ -28,7 +28,8 @@
 SOURCES = schroedinger.ml schroedinger.mli schroedinger_stubs.c ogg_demuxer_schroedinger_decoder.mli ogg_demuxer_schroedinger_decoder.ml
 RESULT = schroedinger
 OCAMLDOCFLAGS = -stars
-LIBINSTALL_FILES = $(wildcard *.mli *.cmi *.cma *.cmxa *.cmx *.a *.so)
+DLL_EXT:=$(shell ocamlfind ocamlc -config | awk '/^ext_dll:/ {print $$2}')
+LIBINSTALL_FILES = $(wildcard *.mli *.cmti *.cmi *.cma *.cmx *.cmxa *.$(EXT_LIB) *$(DLL_EXT))
 ACLIBS = @LIBS@
 LDFLAGS = @LDFLAGS@
 CLIBS = $(ACLIBS:-l%=%)
