diff -c -r conduit.0.13.0/Makefile conduit.0.13.0-fixed/Makefile
*** conduit.0.13.0/Makefile	Mon Sep 19 01:34:03 2016
--- conduit.0.13.0-fixed/Makefile	Tue Oct 18 12:04:35 2016
***************
*** 1,17 ****
  .PHONY: all install clean doc github
  
  PREFIX ?= /usr/local/bin
  
! all: ppx
  	@./build.sh
  
! install: ppx
  	@./build.sh true
  
  clean:
! 	rm -rf _build _install ppx lib/conduit_config.mlh META
  
! doc: ppx
  	@BUILD_DOC=true ./build.sh
  
  github: doc
--- 1,19 ----
+ EXE=.exe
+ 
  .PHONY: all install clean doc github
  
  PREFIX ?= /usr/local/bin
  
! all: ppx$(EXE)
  	@./build.sh
  
! install: ppx$(EXE)
  	@./build.sh true
  
  clean:
! 	rm -rf _build _install ppx$(EXE) lib/conduit_config.mlh META
  
! doc: ppx$(EXE)
  	@BUILD_DOC=true ./build.sh
  
  github: doc
***************
*** 40,45 ****
  	opam publish prepare $(NAME).$(VERSION) $(ARCHIVE)
  	OPAMYES=1 opam publish submit $(NAME).$(VERSION) && rm -rf $(NAME).$(VERSION)
  
! ppx:
! 	ocamlfind ocamlopt -predicates ppx_driver -o ppx -linkpkg \
  	  -package ppx_sexp_conv ppx_driver_runner.cmxa
--- 42,47 ----
  	opam publish prepare $(NAME).$(VERSION) $(ARCHIVE)
  	OPAMYES=1 opam publish submit $(NAME).$(VERSION) && rm -rf $(NAME).$(VERSION)
  
! ppx$(EXE):
! 	ocamlfind ocamlopt -predicates ppx_driver -o ppx$(EXE) -linkpkg \
  	  -package ppx_sexp_conv ppx_driver_runner.cmxa
diff -c -r conduit.0.13.0/build.sh conduit.0.13.0-fixed/build.sh
*** conduit.0.13.0/build.sh	Mon Sep 19 01:34:03 2016
--- conduit.0.13.0-fixed/build.sh	Tue Oct 18 12:07:15 2016
***************
*** 56,62 ****
  
  echo "true: config" >> _tags
  
! echo "true: pp($(pwd)/ppx)" >> _tags
  
  if [ "$HAVE_ASYNC" != "" ]; then
    echo "Building with Async support."
--- 56,63 ----
  
  echo "true: config" >> _tags
  
! PWD=$(cygpath -m `pwd`)
! echo "true: pp($PWD/ppx)" >> _tags
  
  if [ "$HAVE_ASYNC" != "" ]; then
    echo "Building with Async support."
