--- ./configure
+++ ./configure
@@ -64,6 +64,13 @@
     apache=""
     cpp=cpp
     cpp_set=0
+    if [ -n "$CPP" ]; then
+        if which "$CPP" >/dev/null 2>&1 ; then
+            cpp=$CPP
+            cpp_set=1
+        fi
+    fi
+
 }
 
 ocamlc=`get_path ocamlc`
--- ./src/netstring/Makefile.def
+++ ./src/netstring/Makefile.def
@@ -124,11 +124,12 @@
 	(cd netdb && ls *.netdb) | \
 	{ while read f; do echo "$(NET_DB_DIR)/$$f"; done; } >netdb-packlist
 
+NET_DB_DIR_CYGWIN=$(shell cygpath "$(NET_DB_DIR)")
 .PHONY: install-netdb
 install-netdb:
-	mkdir -p "$(DESTDIR)$(NET_DB_DIR)"
-	cp netdb/*.netdb "$(DESTDIR)$(NET_DB_DIR)"
-	@echo "Installed .netdb files into $(DESTDIR)$(NET_DB_DIR)"
+	mkdir -p "$(DESTDIR)$(NET_DB_DIR_CYGWIN)"
+	cp netdb/*.netdb "$(DESTDIR)$(NET_DB_DIR_CYGWIN)"
+	@echo "Installed .netdb files into $(DESTDIR)$(NET_DB_DIR_CYGWIN)"
 
 tools_done:
 	cd tools && $(MAKE)
--- ./src/netstring/netdb.mlp
+++ ./src/netstring/netdb.mlp
@@ -4,7 +4,19 @@
  *)
 
 
-let net_db_dir = ref "@NET_DB_DIR@" ;;
+let net_db_dir = ref (
+  match Sys.os_type with
+  | "Win32" -> 
+    (try 
+      let x = Sys.getenv "OCAMLNET_DB_DIR" in
+      if Sys.is_directory x then
+        x
+      else
+        raise Not_found
+    with
+    | _ -> "@NET_DB_DIR@" )
+  | _ -> "@NET_DB_DIR@" )
+
 
 let net_db_hash = Hashtbl.create 20 ;;
 
--- ./src/netsys/configure
+++ ./src/netsys/configure
@@ -338,8 +338,14 @@
 
 printf "Checking for POSIX clocks..."
 
-cclib=""
-compile_emit clock_gettime HAVE_CLOCK
+if [ $win32 -gt 0 ]; then
+    # For some strange reason the test does not work on win32. So
+    # force to skip this.
+    echo "#undef HAVE_CLOCK" >> config.h
+else
+    cclib=""
+    compile_emit clock_gettime HAVE_CLOCK
+fi
 
 ######################################################################
 
--- ./src/netsys/netsys_c.c
+++ ./src/netsys/netsys_c.c
@@ -75,6 +75,8 @@
 CAMLprim value netsys_sysconf_open_max (value unit) {
 #ifdef HAVE_SYSCONF
     return Val_long(sysconf(_SC_OPEN_MAX));
+#elif _WIN32
+    return Val_long(_getmaxstdio () );
 #else
     invalid_argument("Netsys.sysconf_open_max not available");
 #endif
