--- ./build/Makefile
+++ ./build/Makefile
@@ -30,22 +30,22 @@
 print-%  : ; @echo $* = $($*)
 
 $(SRC:%.mli=%.cmi):%.cmi: $S/%.mli
-	@$(OCAMLC) -c $< -o $@
+	@$(OCAMLC) -c $(shell cygpath -m $<) -o $@
 
 %.cmo: $S/%.ml %.cmi
-	@$(OCAMLC) -c $< -o $@
+	@$(OCAMLC) -c $(shell cygpath -m $<) -o $@
 
 $(OBJ):%.cmx: $S/%.ml
-	@$(OCAMLOPT) -c $< -o $@
+	@$(OCAMLOPT) -c $(shell cygpath -m $<) -o $@
 
 $(BGENS:=.cmx):%.cmx: %.ml %.cmi
-	@$(OCAMLOPT) -c $< -o $@
+	@$(OCAMLOPT) -c $(shell cygpath -m $<) -o $@
 
 lib/sexp/sexp_parse.cmi: lib/sexp/sexp_parse.mli
-	@$(OCAMLC) -c $< -o $@
+	@$(OCAMLC) -c $(shell cygpath -m $<) -o $@
 
 lib/sexp/sexp_lex.cmi: lib/sexp/sexp_lex.ml lib/sexp/sexp_parse.cmi
-	@$(OCAMLC) -c $< -o $@
+	@$(OCAMLC) -c $(shell cygpath -m $<) -o $@
 
 
 clean:
@@ -53,7 +53,7 @@
 
 
 %.cmi: %.mli
-	@$(OCAMLC) -c $< -o $@
+	@$(OCAMLC) -c $(shell cygpath -m $<) -o $@
 
 
 $(ODOC):docs/%.odoc: $S/lib/%.mli
--- ./codept.install
+++ ./codept.install
@@ -1,7 +1,7 @@
 bin: ["codept"]
 lib: ["META"
-     "?codept_ocamlbuild.cma"
-     "?codept_ocamlbuild.cmxa"
+     "?_build/ocamlbuild_plugin/codept_ocamlbuild.cma"
+     "?_build/ocamlbuild_plugin/codept_ocamlbuild.cmxa"
      "?_build/ocamlbuild_plugin/codept_ocamlbuild.cmx"
      "?_build/ocamlbuild_plugin/codept_ocamlbuild.cmi"
      "?_build/ocamlbuild_plugin/codept_ocamlbuild.cmti"
