--- ./src/Makefile.in
+++ ./src/Makefile.in
@@ -35,7 +35,7 @@
 SOURCES = opus_stubs.c opus.mli opus.ml ogg_demuxer_opus_decoder.mli ogg_demuxer_opus_decoder.ml
 RES_CLIB_SUF = _stubs
 RESULT = opus
-LIBINSTALL_FILES = $(wildcard *.mli *.cmi *.cma *.cmxa *.cmx *.a *.so)
+LIBINSTALL_FILES = $(wildcard *.mli *.cmti *.cmi *.cma *.cmx *.cmxa *.$(EXT_LIB) *$(EXT_DLL))
 INCDIRS = @OCAMLOGG_INC@
 ACLIBS = @LIBS@
 CLIBS = $(ACLIBS:-l%=%)
