--- ./src/Makefile.in
+++ ./src/Makefile.in
@@ -28,7 +28,7 @@
 RESULT = duppy
 OCAMLDOCFLAGS = -stars
 DOC_FILES = $(filter %.mli, $(SOURCES)) @PA_DOC@
-LIBINSTALL_FILES = $(wildcard *pa_duppy.cm* *.mli *.cmi *.cma *.cmxa *.cmx *.a *.so)
+LIBINSTALL_FILES = $(wildcard *pa_duppy.cm* *.mli *.cmi *.cma *.cmxa *.cmx *.$(EXT_LIB) *.$(EXT_DLL))
 ACLIBS = @LIBS@
 LDFLAGS = @LDFLAGS@
 CLIBS = $(ACLIBS:-l%=%)
