--- ./Makefile
+++ ./Makefile
@@ -25,7 +25,7 @@
 OBJECTS= $(foreach version,$(VERSIONS), \
 					 $(foreach module,$(MODULES), $(module)_$(version).cmo))
 
-METAQUOTS= $(foreach version,$(VERSIONS),ppx_metaquot_$(version))
+METAQUOTS= $(foreach version,$(VERSIONS),ppx_metaquot_$(version)$(EXE))
 .PHONY: all
 all: ppx_tools_versioned.cma ppx_tools_versioned.cmxa $(METAQUOTS)
 
@@ -61,7 +61,7 @@
 	$(wildcard ppx_tools_versioned.*) \
 	$(OBJECTS:.cmo=.cmi) $(wildcard $(OBJECTS:.cmo=.cmx)) \
 	$(wildcard $(OBJECTS:.cmo=.cmt) $(OBJECTS:.cmo=.cmti)) \
-	$(METAQUOTS)
+	$(METAQUOTS) $(foreach version,$(VERSIONS),ppx_metaquot_$(version).cmt ppx_metaquot_$(version).cmi ppx_metaquot_$(version).cmx ppx_metaquot_$(version).o ppx_metaquot_$(version).ml)
 
 .PHONY: reinstall install uninstall
 
@@ -88,7 +88,7 @@
 
 # Metaquot
 
-ppx_metaquot_%: ppx_tools_versioned.cmxa ast_lifter_%.ml ppx_metaquot_%.ml
+ppx_metaquot_%$(EXE): ppx_tools_versioned.cmxa ast_lifter_%.ml ppx_metaquot_%.ml
 	$(OCAMLOPT) -o $@ $(COMPFLAGS) -linkpkg $^
 
 .PHONY: depend
