*** a/setup.ml	2015-08-18 19:14:45.000000000 +0800
--- b/setup.ml	2015-09-26 21:49:10.785520400 +0800
***************
*** 6879,6891 ****
            conf_custom =
              {
                 pre_command =
!                  [(OASISExpr.EBool true, Some (("config/detect.sh", [])))];
                 post_command =
                   [
                      (OASISExpr.EBool true,
                        Some
!                         (("config/discover.sh",
!                            ["$ocamlc"; "src/config.mlh"; "src/config.h"])))
                   ]
              };
            build_type = (`Build, "ocamlbuild", Some "0.4");
--- 6879,6891 ----
            conf_custom =
              {
                 pre_command =
!                  [(OASISExpr.EBool true, Some (("sh", [ "-c"; "\"config/detect.sh\""])))];
                 post_command =
                   [
                      (OASISExpr.EBool true,
                        Some
!                         (("sh",
! 			  [ "-c"; "\"config/discover.sh $ocamlc src/config.mlh src/config.h\"" ])))
                   ]
              };
            build_type = (`Build, "ocamlbuild", Some "0.4");
***************
*** 6915,6921 ****
                 pre_command =
                   [
                      (OASISExpr.EBool true,
!                       Some (("$rm", ["src/config.mlh"; "src/config.h"])))
                   ];
                 post_command = [(OASISExpr.EBool true, None)]
              };
--- 6915,6921 ----
                 pre_command =
                   [
                      (OASISExpr.EBool true,
!                       Some (("sh", [ "-c"; "\"$rm src/config.mlh src/config.h\""])))
                   ];
                 post_command = [(OASISExpr.EBool true, None)]
              };
*** a/src/bigstring_stubs.c	2015-08-18 19:14:45.000000000 +0800
--- b/src/bigstring_stubs.c	2015-09-27 01:32:27.699280000 +0800
***************
*** 31,37 ****
  #define bswap_64 swap64
  #else
  #include <sys/types.h>
! #include <sys/endian.h>
  #define __BYTE_ORDER    _BYTE_ORDER
  #define __LITTLE_ENDIAN _LITTLE_ENDIAN
  #define __BIG_ENDIAN    _BIG_ENDIAN
--- 31,39 ----
  #define bswap_64 swap64
  #else
  #include <sys/types.h>
! // MINGW
! // #include <sys/endian.h>
! #include <sys/param.h>
  #define __BYTE_ORDER    _BYTE_ORDER
  #define __LITTLE_ENDIAN _LITTLE_ENDIAN
  #define __BIG_ENDIAN    _BIG_ENDIAN
