--- ./Makefile
+++ ./Makefile
@@ -96,7 +96,7 @@
  $(strip $(foreach f, $(1), $(if $(filter $(f),$(2)),,$f)))
 endef
 
-export MLEXTRAFILES := $(GENMLFILES) $(GENML4FILES) $(GENPLUGINSMOD)
+export MLEXTRAFILES := $(GENMLFILES)
 export MLSTATICFILES := $(call diff, $(EXISTINGML), $(MLEXTRAFILES))
 export MLIFILES := $(sort $(GENMLIFILES) $(EXISTINGMLI))
 
--- ./Makefile.build
+++ ./Makefile.build
@@ -39,7 +39,8 @@
 MLFILES:=$(MLSTATICFILES) $(MLEXTRAFILES)
 
 ALLDEPS:=$(addsuffix .d, \
-  $(ML4FILES) $(MLFILES) $(MLIFILES) $(CFILES) $(MLLIBFILES) $(VFILES))
+  $(MLFILES) $(MLIFILES) $(CFILES) $(MLLIBFILES))
+
 
 .SECONDARY: $(ALLDEPS) $(GENFILES) $(ML4FILES:.ml4=.ml)
 
@@ -327,13 +328,13 @@
 coqide-opt:  $(COQIDEBYTE) $(COQIDEOPT) $(COQIDE)
 coqide-files: $(IDEFILES)
 
-$(COQIDEOPT): $(LINKIDEOPT) | $(COQTOPOPT)
+$(COQIDEOPT): $(LINKIDEOPT) | $(LIBCOQRUN)
 	$(SHOW)'OCAMLOPT -o $@'
 	$(HIDE)$(OCAMLOPT) $(COQIDEFLAGS) $(OPTFLAGS) -o $@ unix.cmxa threads.cmxa \
 		lablgtk.cmxa $(IDEOPTFLAGS) gtkThread.cmx str.cmxa $(LINKIDEOPT)
 	$(STRIP) $@
 
-$(COQIDEBYTE): $(LINKIDE) | $(COQTOPBYTE)
+$(COQIDEBYTE): $(LINKIDE) | $(LIBCOQRUN)
 	$(SHOW)'OCAMLC -o $@'
 	$(HIDE)$(OCAMLC) $(COQIDEFLAGS) $(BYTEFLAGS) -o $@ unix.cma threads.cma lablgtk.cma gtkThread.cmo\
 		str.cma $(COQRUNBYTEFLAGS) $(LINKIDE)
--- ./Makefile.common
+++ ./Makefile.common
@@ -269,7 +269,7 @@
 
 ## we now retrieve the names of .vo file to compile in */vo.itarget files
 
-cat_vo_itarget = $(addprefix $(1)/,$(shell cat $(1)/vo.itarget))
+cat_vo_itarget =
 
 ## Theories
 
