*** Makefile.orig	2015-09-20 18:49:47.015036500 +0800
--- Makefile	2015-09-20 18:50:07.234221300 +0800
***************
*** 57,72 ****
  	cat findlib.conf.in | \
  		tools/patch '@SITELIB@' '$(OCAML_SITELIB)' >findlib.conf
  	if ocamlc.opt >/dev/null 2>&1; then \
! 		echo 'ocamlc="ocamlc.opt"' >>findlib.conf; \
  	fi
  	if ocamlopt.opt >/dev/null 2>&1; then \
! 		echo 'ocamlopt="ocamlopt.opt"' >>findlib.conf; \
  	fi
  	if ocamldep.opt >/dev/null 2>&1; then \
! 		echo 'ocamldep="ocamldep.opt"' >>findlib.conf; \
  	fi
  	if ocamldoc.opt >/dev/null 2>&1; then \
! 		echo 'ocamldoc="ocamldoc.opt"' >>findlib.conf; \
  	fi
  
  .PHONY: install-doc
--- 57,72 ----
  	cat findlib.conf.in | \
  		tools/patch '@SITELIB@' '$(OCAML_SITELIB)' >findlib.conf
  	if ocamlc.opt >/dev/null 2>&1; then \
! 		echo 'ocamlc="ocamlc.opt.exe"' >>findlib.conf; \
  	fi
  	if ocamlopt.opt >/dev/null 2>&1; then \
! 		echo 'ocamlopt="ocamlopt.opt.exe"' >>findlib.conf; \
  	fi
  	if ocamldep.opt >/dev/null 2>&1; then \
! 		echo 'ocamldep="ocamldep.opt.exe"' >>findlib.conf; \
  	fi
  	if ocamldoc.opt >/dev/null 2>&1; then \
! 		echo 'ocamldoc="ocamldoc.opt.exe"' >>findlib.conf; \
  	fi
  
  .PHONY: install-doc
