--- ./src/unix/config/discover.ml
+++ ./src/unix/config/discover.ml
@@ -342,12 +342,16 @@
     Sys.command s = 0) fmt in
   let obj_file = Filename.chop_suffix stub_file ".c" ^ !ext_obj
   in
+  let obj_flag = match !system with
+  | "win32" | "win64" -> "-ccopt -Fo:"
+  | _ -> "-ccopt -o -ccopt " in
   (* First compile the .c file using -ocamlc and CFLAGS (opt) *)
   (cmd
-    "%s -c %s %s -ccopt -o -ccopt %s >> %s 2>&1"
+    "%s -c %s %s %s%s >> %s 2>&1"
     !ocamlc
     (String.concat " " (List.map (fun x -> "-ccopt " ^ x) (List.map Filename.quote opt)))
     (Filename.quote stub_file)
+    obj_flag
     (Filename.quote obj_file)
     (Filename.quote !log_file))
   &&
