diff -pur8 sqlite3.4.0.2.old/lib/sqlite3_stubs.c sqlite3.4.0.2/lib/sqlite3_stubs.c
--- sqlite3.4.0.2.old/lib/sqlite3_stubs.c	2016-01-05 13:41:46.619655400 +0100
+++ sqlite3.4.0.2/lib/sqlite3_stubs.c	2016-01-05 13:42:18.252612800 +0100
@@ -44,17 +44,17 @@
 # if !defined(__FreeBSD__) && !__APPLE__
 # define __unused __attribute__ ((unused))
 # endif
 #else
 # define __unused
 # define inline
 #endif
 
-#if SQLITE_VERSION_NUMBER >= 3003007
+#if false
 # define SQLITE_HAS_ENABLE_LOAD_EXTENSION
 #endif
 
 #if SQLITE_VERSION_NUMBER >= 3003009
 # define my_sqlite3_prepare sqlite3_prepare_v2
 #else
 # define my_sqlite3_prepare sqlite3_prepare
 #endif
