--- ./config/ciconfig.ml
+++ ./config/ciconfig.ml
@@ -42,7 +42,7 @@
 
 let xpm = find_library
     [ by_pkg_config "xpm"
-    ; by_cc ~c_flags:[] ~link_flags:["-lXpm"] ~headers:["X11/xpm.h"] ~functions:["XpmReadFileToXpmImage"]
+    ; by_cc ~c_flags:[] ~link_flags:["-lXpm"] ~headers:["inttypes.h"; "X11/xpm.h"] ~functions:["XpmReadFileToXpmImage"]
     ]
 
 let rgb_txt = find_file "rgb.txt"
--- ./xpm/xpmread.c
+++ ./xpm/xpmread.c
@@ -16,13 +16,16 @@
 
 #ifdef HAS_XPM
 
+#ifdef _WIN32
+#include <inttypes.h>
+#endif
+#include <X11/xpm.h>
+
 #include <caml/mlvalues.h>
 #include <caml/alloc.h>
 #include <caml/memory.h>
 #include <caml/fail.h>
 
-#include <X11/xpm.h>
-
 #include <stdio.h>
 
 value read_xpm_file( name )
